flow360/component/simulation/operating_condition.py

```diff
@@ -1,50 +1,117 @@
-    1. The physical (non-geometrical) reference values for the problem.
-    2. The initial condition for the problem.
-
-    TODO:
-    1. What other types of operation conditions do we need?
-"""
-
-
-class TurbulenceQuantities(Flow360BaseModel):
-    """PLACE HOLDER, Should be exactly the the same as `TurbulenceQuantitiesType` in current Flow360Params"""
-
-    pass
-
-
-class ExternalFlowOperatingConditions(Flow360BaseModel):
-    Mach: float = pd.Field()
-    alpha: float = pd.Field(0)
-    beta: float = pd.Field(0)
-    temperature: float = pd.Field(288.15)
-    reference_velocity: Optional[float] = pd.Field()  # See U_{ref} definition in our documentation
-
-    initial_flow_condition: Optional[tuple[str, str, str, str, str]] = pd.Field(
-        ("NotImplemented", "NotImplemented", "NotImplemented")
-    )
-    turbulence_quantities = Optional[TurbulenceQuantities] = pd.Field()
-
-
-class InternalFlowOperatingConditions(Flow360BaseModel):
-    pressure_difference: float = pd.Field()
-    reference_velocity: float = pd.Field()
-    inlet_velocity: float = pd.Field()
-
-
-class SolidOperatingConditions(Flow360BaseModel):
-    initial_temperature: float = pd.Field()
-
-
-OperatingConditionTypes = Union[
-    ExternalFlowOperatingConditions, InternalFlowOperatingConditions, SolidOperatingConditions
-]
+VelocityVectorType = Unioin[VelocityType.Vector, Tuple[pd.StrictStr, pd.StrictStr, pd.StrictStr]]
+
+
+class ThermalCondition(Flow360BaseModel):
+    temperature: TemperatureType = 288.15
+    density: DensityType = 1.225
+    material: materialTypes = Air()
+    _altitude: Optional[LengthType] = None
+    _temperature_offset: Optional[TemperatureType] = None
+
+    @validate_arguments
+    @classmethod
+    def from_standard_atmosphere(
+        cls, altitude: LengthType = 0, temperature_offset: TemperatureType = 0
+    ):
+        # TODO: add standard atmosphere implementation
+        density = 1.225
+        temperature = 288.15
+
+        return cls(
+            density=density,
+            temperature=temperature,
+            material=Air(),
+        )
+
+    @computed_field
+    @property
+    def altitude(self) -> LengthType:
+        return self._altitude
+
+    @computed_field
+    @property
+    def temperature_offset(self) -> TemperatureType:
+        return self._temperature_offset
+
+    @property
+    def speed_of_sound(self) -> VelocityType:
+        return np.sqrt(
+            self.material.specific_heat_ratio * self.material.gas_constant * self.temperature
+        )
+
+    @property
+    def pressure(self) -> PressureType:
+        # TODO: implement
+        return 1.013e5
+
+    @property
+    def dynamic_viscosity(self) -> ViscosityType:
+        # TODO: implement
+        return 1.825e-5
+
+
+class GenericOperatingCondition(Flow360BaseModel):
+    """
+    Operating condition defines the physical (non-geometrical) reference values for the problem.
+    """
+
+    speed: VelocityType.Positive
```

Comment:
may be call it reference_speed? User might get confused what speed this is?

Reply:
OperatingCondition itself is supposed to be reference. I'm not against changing the name, but thermal_condition should be reference_thermal_condition as well